internal/testrunner/runners/system/runner.go

```diff
@@ -478,6 +479,17 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 		}
 	}
 
```

Comment:
When this new assertion is used the condition inside of `waitUntilTrue` should be changed. It should continue searching until it reaches at least the expected count.

Reply:
@andrewkroh 

The assertion validates how many docs were harvested, asking "Did we get X docs?"

Continuing to search until a determined number of documents have been acquired feels to me more like setting a target, requesting "Go get X docs?"

These could resolve to the same value in many cases, which could make the assertion itself redundant.

I can see 2 ways to do this, either add an additional parameter, so:

```
assert:
  hit_count: 3
target:
  doc_count: 3
```

or alternatively rename `assert` to `target`, so 


```
target:
  doc_count: 3
```

or maybe 

```
target:
  hit_count: 3
```

What do you think?

Reply:
Moving back to review pending feedback from @andrewkroh on this remaining issue

Reply:
Hi @andrewkroh would you prefer the additional parameter suggested above, or to rename? Thanks

Reply:
sumarising offline conversation fine to keep

```
assert:
  hit_count: 3
```